crates/uv-git/src/git.rs

```diff
@@ -802,27 +788,23 @@ fn github_fast_path(
         }
 
         let response = request.send().await?;
+
+        // GitHub returns a 404 if the repository does not exist, and a 422 if it exists but GitHub
+        // is unable to resolve the requested revision.
         response.error_for_status_ref()?;
+
         let response_code = response.status();
         if response_code == StatusCode::NOT_MODIFIED {
             Ok(FastPathRev::UpToDate)
         } else if response_code == StatusCode::OK {
             let oid_to_fetch = response.text().await?.parse()?;
             Ok(FastPathRev::NeedsFetch(oid_to_fetch))
         } else {
-            // Usually response_code == 404 if the repository does not exist, and
-            // response_code == 422 if exists but GitHub is unable to resolve the
-            // requested rev.
```

Comment:
This is in the wrong place. 400-level errors are raised by `response.error_for_status_ref()?`!